lib/logstash/inputs/kafka.rb

```diff
@@ -75,23 +84,30 @@ class LogStash::Inputs::Kafka < LogStash::Inputs::Base
   public
   def register
     LogStash::Logger.setup_log4j(@logger)
-    require 'jruby-kafka'
-
     options = {
         :zk_connect => @zk_connect,
         :group_id => @group_id,
         :topic_id => @topic_id,
+        :auto_offset_reset => @auto_offset_reset,
         :rebalance_max_retries => @rebalance_max_retries,
         :rebalance_backoff_ms => @rebalance_backoff_ms,
         :consumer_timeout_ms => @consumer_timeout_ms,
         :consumer_restart_on_error => @consumer_restart_on_error,
         :consumer_restart_sleep_ms => @consumer_restart_sleep_ms,
         :consumer_id => @consumer_id,
-        :fetch_message_max_bytes => @fetch_message_max_bytes
+        :fetch_message_max_bytes => @fetch_message_max_bytes,
+        :allow_topics => @white_list,
+        :filter_topics => @black_list
     }
     if @reset_beginning
       options[:reset_beginning] = 'from-beginning'
     end # if :reset_beginning
+    topic_or_filter = [@topic_id, @white_list, @black_list].compact
+    if topic_or_filter.count == 0
+      raise('topic_id, white_list or black_list required.')
```

Comment:
also, can we raise a LogStash::ConfigurationError here?

and the following `raise`